### ISSUE TITLE
PHP 8.1 constant property

### ISSUE THREAD

**Issue by lloricode** [slevomat/coding-standard#1331]
Hi!,

In php8.1, overriding constant propery in child class is allowed, so using `static::PROPERTY` is has a reason to do,
Notice of changes in calling constant property, 

this changes was introduce bugs on our application
```php
abstract class BaseAPI
{
    protected const VERSION = 1;
```

![Screenshot from 2022-01-26 07-53-30](https://user-images.githubusercontent.com/8251344/151079304-d8d8dca7-3dc9-4188-a5a0-2fed5fd9c1bb.png)

PHP: 8.1.0
using library: https://github.com/nunomaduro/phpinsights v2.1.0

Thank you

**Comment by kukulich** [comment]
@lloricode I’m sorry I don’t understand your report. Where’s the bug?

**Comment by lloricode** [comment]
> @lloricode I’m sorry I don’t understand your report. Where’s the bug?

the given image should not change, its intended to use `static::`

**Comment by kukulich** [comment]
@lloricode What sniff caused the change?

**Comment by lloricode** [comment]
im not directly use this package, im just using library: https://github.com/nunomaduro/phpinsights v2.1.0, if i made issue there, they recommend me to report here, i let you know if i can give you the details

**Comment by kukulich** [comment]
Marked currently as invalid.

**Comment by lloricode** [comment]
> @lloricode What sniff caused the change?

Hi, the issue comming from this [sniff](https://github.com/slevomat/coding-standard/blob/master/SlevomatCodingStandard/Sniffs/Classes/DisallowLateStaticBindingForConstantsSniff.php)
